pipelines/templates/tasks/unitybuild.yml

```diff
@@ -123,6 +123,14 @@ steps:
    }
   displayName: "Build ${{ parameters.Platform }} ${{ parameters.Arch }} ${{ parameters.ScriptingBackend }}"
 
+- task: PowerShell@2
+  displayName: 'Validate build logs'
+  inputs:
+    targetType: filePath
+    filePath: ./scripts/ci/validatebuildlog.ps1
+    arguments: >
+      -LogFile: '$(Build.ArtifactStagingDirectory)\build\${{ parameters.Platform }}_${{ parameters.Arch }}_${{ parameters.ScriptingBackend }}\build\build.log'

```

Comment:
It might be nice to add logDirectory and logPath as task parameters (with default values). This would ensure that the log path is always the same in unity build task and any log processing tasks that follow.
There's a way to emit a value from a task and reuse it (using ##vso tags) but I'm not sure how to use it effectively (something I need to experiment a bit more with) and parameters are easy to use.

Reply:
This is a good idea - I'm going to look into this after getting this change in, if only because I will need a little more time to figure out how to refactor this stuff to make the string interpolation work with other parameters (in the param section).